lib/cn/cn.c

```diff
@@ -930,27 +965,20 @@ cn_open(
         vcnt = atomic_read(&cn_kvdb->cnd_vblk_cnt);
     }
 
-    ctx.cn = cn;
-    ctx.nodemap = map_create(CN_FANOUT_MAX);
-    if (ev(!ctx.nodemap)) {
-        err = merr(ENOMEM);
+    /* Build tree from data in CNDB.
+     */
+    err = cndb_cn_ctx_init(&ctx, cn->cn_tree, cn->cn_tree->ct_root);
+    if (ev(err))
         goto err_exit;
-    }
 
-    /* cn_tree_create() always creates the root node and appends it to the
-     * list of tree nodes (i.e., tree->ct_nodes).  cn_kvset_cb() will create
-     * all leaf nodes and append them to the list in the rehydration order
-     * specified by cndb_cn_instantiate() (i.e., no particular order).
-     */
-    map_insert_ptr(ctx.nodemap, 0, cn->cn_tree->ct_root);
+    err = cndb_cn_instantiate(cndb, cnid, &ctx, cndb_cn_callback);
+
+    cndb_cn_ctx_fini(&ctx);
 
-    err = cndb_cn_instantiate(cndb, cnid, &ctx, cn_kvset_cb);
     if (ev(err))
         goto err_exit;
 
-    /* Walk the list of nodes created/populated by cn_kvset_cb() and insert
-     * all leaf nodes into the route map (i.e., all nodes except the root
-     * node which always has node ID 0).
```

Comment:
I think this is an pretty informative comment for someone wondering how the list got built...

Reply:
Will revert.